Sources/_CryptoExtras/RSA/RSA_boring.swift

```diff
@@ -315,6 +344,215 @@ extension BoringSSLRSAPublicKey {
+            /// subsequent calls will return NULL as well, so it is sufficient to check the return value of the last
+            /// BN_CTX_get() call.
+            /// ```
+            /// —— Extract from `man 3 BN_CTX_get`.
+            let bnCtx = CCryptoBoringSSL_BN_CTX_new()
+            CCryptoBoringSSL_BN_CTX_start(bnCtx)
+            defer {
+                CCryptoBoringSSL_BN_CTX_end(bnCtx)
+                CCryptoBoringSSL_BN_CTX_free(bnCtx)
+            }
+            let m = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let gcd = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let r = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let inv = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let x = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            let z = CCryptoBoringSSL_BN_CTX_get(bnCtx)
+            guard z != nil else { throw CryptoKitError.internalBoringSSLError() }
+
+            let rsaPublicKey = CCryptoBoringSSL_EVP_PKEY_get0_RSA(self.pointer)
+            let outputSize = Int(CCryptoBoringSSL_RSA_size(rsaPublicKey))
+
+            // 1. encoded_msg = EMSA-PSS-ENCODE(msg, bit_len(n)) with Hash, MGF, and salt_len as defined in the parameters
+            // 2. If EMSA-PSS-ENCODE raises an error, re-raise the error and stop
+            let hashDigestType = try! DigestType(forDigestType: H.Digest.self)
+            let hash = H.hash(data: message.rawRepresentation)
+            let saltLength: Int32
+            switch parameters.padding {
+            case .PSS: saltLength = Int32(H.Digest.byteCount)
+            case .PSSZERO: saltLength = 0
+            }
+            var encodedMessage = [UInt8](repeating: 0, count: outputSize)
+            guard hash.withUnsafeBytes({ hashBufferPtr in
+                CCryptoBoringSSL_RSA_padding_add_PKCS1_PSS_mgf1(
+                    rsaPublicKey,
+                    &encodedMessage,
+                    hashBufferPtr.baseAddress,
+                    hashDigestType.dispatchTable,
+                    hashDigestType.dispatchTable,
+                    saltLength
+                )
+            }) == 1 else {
+                switch ERR_GET_REASON(CCryptoBoringSSL_ERR_get_error()) {
+                case RSA_R_DATA_TOO_LARGE_FOR_KEY_SIZE:
+                    throw _RSA.BlindSigning.Error.messageTooLong
+                default:
+                    throw CryptoKitError.internalBoringSSLError()
+                }
+            }
+
+            // 3. m = bytes_to_int(encoded_msg)
+            CCryptoBoringSSL_BN_bin2bn(&encodedMessage, encodedMessage.count, m)
+
+            // 4. c = is_coprime(m, n)
+            let n = CCryptoBoringSSL_RSA_get0_n(rsaPublicKey)
+            CCryptoBoringSSL_BN_gcd(gcd, m, n, bnCtx)
+            let c = CCryptoBoringSSL_BN_is_one(gcd) == 1
+
+            // 5. If c is false, raise an "invalid input" error and stop
+            guard c else {
```

Comment:
Given the text above, it might be better to write this to exactly match: `if !c`